make/test.mk

```
@@ -333,13 +333,12 @@ create-host-resources:
 	-oc create -f deploy/host-operator/${ENVIRONMENT}/ -n ${HOST_NS}
 	# patch toolchainconfig to prevent webhook deploy for 2nd member, a 2nd webhook deploy causes the webhook verification in e2e tests to fail
 	# since e2e environment has 2 member operators running in the same cluster
+	# for details on how the TOOLCHAINCLUSTER_NAME is composed see https://github.com/codeready-toolchain/toolchain-cicd/blob/master/scripts/add-cluster.sh
 	if [[ ${SECOND_MEMBER_MODE} == true ]]; then \
-		API_ENDPOINT=`oc get infrastructure cluster -o jsonpath='{.status.apiServerURL}'`; \
-		TOOLCHAIN_CLUSTER_NAME=`echo "$${API_ENDPOINT}" | sed 's/.*api\.\([^:]*\):.*/\1/'`; \
-		echo "API_ENDPOINT $${API_ENDPOINT}"; \
+		TOOLCHAIN_CLUSTER_NAME=`oc get toolchaincluster -l type=member -A --no-headers -o custom-columns=":metadata.name" | grep -E "2\>"`; \
```

we need to list only in host operator ns, plus I would filter only  should get only the `tenant` ToolchainCluster CRs - just in case someone would rerun the target against an already existing deployment/namespaces that would already contain the e2e ToolchainCluster CRs.
and as for the grep - doesn't it select a name that has `2` anywhere in the name? shouldn't it be:
```suggestion
		TOOLCHAIN_CLUSTER_NAME=`oc get toolchaincluster -l type=member,cluster-role.toolchain.dev.openshift.com/tenant=""  -n ${HOST_NS} --no-headers -o custom-columns=":metadata.name" | grep "2$"`; \
```

---

Good point about adding the namespace.

About the `cluster-role.toolchain.dev.openshift.com/tenant=""` I'm not sure it's 100% required , if you are referring to `e2e-fmuntean-devsandbox-cluster1.devcluster.openshift.com2` type of `toolchaincluster` they do not have the label `type=member` but `type=e2e` :

```
 labels:
    namespace: toolchain-member-14130231
    ownerClusterName: member-fmuntean-devsandbox-cluster1.devcluster.openshift.com
    type: e2e
```
but probably I've misunderstood.

As per the `grep` part seems to be working the same way:
```
✗ oc get toolchaincluster -l type=member  -n toolchain-host-14130231 --no-headers -o custom-columns=":metadata.name" | grep "2$"
member-fmuntean-devsandbox-cluster1.devcluster.openshift.com2

✗ oc get toolchaincluster -l type=member  -n toolchain-host-14130231 --no-headers -o custom-columns=":metadata.name" | grep -E "2\>"
member-fmuntean-devsandbox-cluster1.devcluster.openshift.com2
```
 I can change it since your version seems more intuitive probably.

---

> About the cluster-role.toolchain.dev.openshift.com/tenant="" I'm not sure it's 100% required , if you are referring to e2e-fmuntean-devsandbox-cluster1.devcluster.openshift.com2 type of toolchaincluster they do not have the label type=member but type=e2e :

you are right :+1: , my bad :man_facepalming: I forgot that we use different label

as for grep:
```bash
$ echo "member-sandbox.openshiftapps.com2
member-sandbox-m2.openshiftapps.com" | grep -E "2\>"
member-sandbox.openshiftapps.com2
member-sandbox-m2.openshiftapps.com
```
vs
```bash
$ echo "member-sandbox.openshiftapps.com2
member-sandbox-m2.openshiftapps.com" | grep -E "2$"
member-sandbox.openshiftapps.com2
```

---

I thought I tested that, good catch! 
I've hopefully reflected all the required changes in [29fe04d](https://github.com/codeready-toolchain/toolchain-e2e/pull/677/commits/29fe04dc3d43f0f5d04fca32f5186f221c963cdd)

Thanks for great review!

---

hey @MatousJobanek I now remember why I ended up using `grep -E 2\>` instead of  `grep 2$` command. 
It's because was not working initially in make target, since it needs the extra $ simbol 😕 

Please see: [5927024](https://github.com/codeready-toolchain/toolchain-e2e/pull/677/commits/5927024164a577226c4d58533f6194871e2f0cee)